packages/flutter/lib/src/gestures/force_press.dart

```diff
@@ -0,0 +1,255 @@
+  /// of the screen, pressing the screen with varying forces or both
+  /// simultaneously.
+  ///
+  /// This callback will be invoked for every frame after the invocation of
+  /// [onStart] and before the invocation of [onEnd], no matter what the
+  /// pressure is during this time period. The position and pressure of the
+  /// pointer is provided in the callback's `details` argument, which is a
+  /// [ForcePressUpdateDetails] object.
+  GestureForcePressUpdateCallback onUpdate;
+
+  /// A pointer is in contact with the screen and has just pressed with a force
+  /// exceeding the [peakPressure]. This is an arbitrary second level action
+  /// threshold and isn't necessarily the maximum possible device pressure
+  /// (which is 1.0).
+  ///
+  /// The position of the pointer is provided in the callback's `details`
+  /// argument, which is a [ForcePressDetails] object.
+  GestureForcePressPeakCallback onPeak;
+
+  /// A pointer is no longer in contact with the screen.
+  ///
+  /// The position of the pointer is provided in the callback's `details`
+  /// argument, which is a [ForcePressDetails] object.
+  GestureForcePressEndCallback onEnd;
+
+  /// The pressure of the press required to initiate a force press.
+  ///
+  /// A value of 0.0 is no pressure, and 1.0 is maximum pressure.
+  double startPressure;
+
+  /// The pressure of the press required to peak a force press.
+  ///
+  /// A value of 0.0 is no pressure, and 1.0 is maximum pressure. This value
+  /// must be greater than [startPressure].
+  double peakPressure;
+
+  Offset _lastPosition;
+  double _lastPressure;
+  _ForceState _state = _ForceState.ready;
+
+  @override
+  void addPointer(PointerEvent event) {
+    startTrackingPointer(event.pointer);
+    if (_state == _ForceState.ready) {
+      _state = _ForceState.possible;
+      _lastPosition = event.position;
+    }
+  }
+
+  @override
+  void handleEvent(PointerEvent event) {
+    assert(_state != _ForceState.ready);
+    // A static pointer with changes in pressure creates PointerMoveEvent events.
+    if (event is PointerMoveEvent || event is PointerDownEvent) {
+      final double pressure = _inverseLerp(event.pressureMin, event.pressureMax, event.pressure);
+      _lastPosition = event.position;
+      _lastPressure = pressure;
+
+      if (_state == _ForceState.possible) {
+        if (pressure > startPressure) {
+          _state = _ForceState.started;
+          resolve(GestureDisposition.accepted);
+        } else if (event.delta.distanceSquared > kTouchSlop)
+          resolve(GestureDisposition.rejected);
+      }
+      // In case this is the only gesture detector we still don't want to start
+      // the gesture until the pressure is greater than the startPressure.
+      if (pressure > startPressure && _state == _ForceState.accepted) {
+        _state = _ForceState.started;
+        invokeCallback<void>('onStart', () => onStart(ForcePressDetails(
```

Comment:
this implies we'll call onStart twice, once when you go from possible to accepted, and once from accepted to started.

Reply:
oh never mind, i misread the code below.